lib/widgets/lightbox.dart

```diff
@@ -0,0 +1,237 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:intl/intl.dart';
+
+import '../api/model/model.dart';
+import 'content.dart';
+import 'page.dart';
+import 'clipboard.dart';
+import 'store.dart';
+
+// TODO: Add index of the image preview in the message, to not break if
+//   there are multiple image previews with the same URL in the same
+//   message. Maybe keep `src`, so that on exit the lightbox image doesn't
+//   fly to an image preview with a different URL, following a message edit
+//   while the lightbox was open.
+class _LightboxHeroTag {
+  _LightboxHeroTag({required this.messageId, required this.src});
+
+  final int messageId;
+  final String src;
+
+  @override
+  bool operator ==(Object other) {
+    return other is _LightboxHeroTag &&
+      other.messageId == messageId &&
+      other.src == src;
+  }
+
+  @override
+  int get hashCode => Object.hash('_LightboxHeroTag', messageId, src);
+}
+
+/// Builds a [Hero] from an image in the message list to the lightbox page.
+class LightboxHero extends StatelessWidget {
+  const LightboxHero({
+    super.key,
+    required this.message,
+    required this.src,
+    required this.child,
+  });
+
+  final Message message;
+  final String src;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Hero(
+      tag: _LightboxHeroTag(messageId: message.id, src: src),
+      flightShuttleBuilder: (
+        BuildContext flightContext,
+        Animation<double> animation,
+        HeroFlightDirection flightDirection,
+        BuildContext fromHeroContext,
+        BuildContext toHeroContext,
+      ) {
+        final accountId = PerAccountStoreWidget.accountIdOf(fromHeroContext);
+        return PerAccountStoreWidget(
+          accountId: accountId,
+          child: RealmContentNetworkImage(src, filterQuality: FilterQuality.medium));
+      },
+      child: child,
```

Comment:
And in fact this feels like it's encoding an assumption that the `child` field is basically the same as this RealmContentNetworkImage widget. So it may as well construct it itself in all cases. Like this, say:
```dart
  Widget build(BuildContext context) {
    final child = RealmContentNetworkImage(src, filterQuality: FilterQuality.medium);
    return Hero(
      // …
          child: child);
      },
      child: child);
```